source/common/upstream/subset_lb.h

```diff
@@ -178,6 +186,9 @@ class SubsetLoadBalancer : public LoadBalancer, Logger::Loggable<Logger::Id::ups
 
   // Forms a trie-like structure. Requires lexically sorted Host and Route metadata.
   LbSubsetMap subsets_;
+  // Forms a trie-like structure of lexically sorted keys+optional fallback policy from subset
+  // selectors configuration
+  SubsetSelectorMap selectors_;
```

Comment:
What's the reason you need to maintain a separate trie for this? Wouldn't the structure of this match `subsets_` in such a way that you could embed the new fallback policy into the nodes of that?